src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+  }
+
+  /**
+   * @brief Get the element type for the meshes
+   * 
+   * @return integer 
+   */
+  integer GetElemType() const { return elem_type_; }
+
+  /**
+   * @brief Get pointers to component arrays of the coordinates on the redecomp mesh
+   * 
+   * @return std::vector<const real*> of length 3
+   *
+   * @note The third entry is nullptr in two dimensions
+   */
+  std::vector<const real*> GetRedecompCoordsPtrs() const 
+  { 
+    return coords_.GetRedecompFieldPtrs();
+  }
+
+  /**
+   * @brief Get pointers to component arrays of the nodal response on the redecomp mesh
+   * 
+   * @return std::vector<real*> of length 3
+   *
+   * @note The third entry is nullptr in two dimensions
+   */
+  std::vector<real*> GetRedecompResponsePtrs()
+  {
+    return ParentField::GetRedecompFieldPtrs(redecomp_response_);
+  }
+
+  /**
+   * @brief Get the nodal response grid function on the redecomp mesh
+   * 
+   * @return const mfem::GridFunction& 
+   */
+  const mfem::GridFunction& GetRedecompResponse() const
+  {
+    return redecomp_response_;
+  }
+
+  /**
+   * @brief Get the nodal response grid function on the parent mesh
+   * 
+   * @return mfem::ParGridFunction 
+   */
+  mfem::ParGridFunction GetParentResponse() const
+  {
+    return GetParentTransfer().RedecompToParent(redecomp_response_);
+  }
+
+  /**
+   * @brief Get the parent to redecomp transfer object
+   * 
+   * @return const ParentRedecompTransfer& 
+   */
+  const ParentRedecompTransfer& GetParentTransfer() const
+  { 
+    return GetUpdateData().vector_xfer_;
+  }
+
+  /**
+   * @brief Add/replace the parent velocity grid function
+   * 
+   * @param velocity Velocity grid function on the parent mesh
+   */
+  void SetParentVelocity(const mfem::ParGridFunction& velocity);
+
+  /**
+   * @brief Determine if a velocity grid function has been set
+   * 
+   * @return true: Velocity grid function has been set
+   * @return false: Velocity grid function has not been set
+   */
+  bool HasVelocity() const { return velocity_ != nullptr; }
+
+  /**
+   * @brief Get pointers to component arrays of the velocity on the redecomp mesh
+   * 
+   * @return std::vector<const real*> of length 3
+   *
+   * @note The third entry is nullptr in two dimensions
+   */
+  std::vector<const real*> GetRedecompVelocityPtrs() const
+  {
+    return velocity_->GetRedecompFieldPtrs();
+  }
+
+  /**
+   * @brief Get the map from mesh 1 element indices to redecomp mesh element
+   * indices
+   *
+   * @return const std::vector<integer>& 
+   */
+  const std::vector<integer>& GetElemMap1() const
+  {
+    return GetUpdateData().elem_map_1_;
+  }
+
+  /**
+   * @brief Get the map from mesh 2 element indices to redecomp mesh element
+   * indices
+   *
+   * @return const std::vector<integer>& 
+   */
+  const std::vector<integer>& GetElemMap2() const
+  {
+    return GetUpdateData().elem_map_2_;
+  }
+
+  /**
+   * @brief Get the (coarse) boundary submesh containing contact surfaces
```

Comment:
Wording is confusing. Is there a single submesh for each of the two contact surfaces in a coupling scheme, or are there two submeshes? At some point this was not made clear and I missed it. Consider clarifying where appropriate.

Reply:
There is one `mfem::SubMesh` so Tribol can use the DOF numbering on the `RedecompMesh`.  Otherwise we would need to define a DOF offset and a mapping between DOF numbers.  I think this is specific to the mortar method Jacobian numbering; two submeshes would work fine for common plane, but one submesh is more general.